### PR TITLE
Added DET/OVERLAP option to user file parser

### LINKED ISSUE #21776: Fit range selection updates
Steve requested the following changes to the new merge fit range functionality. 

- Rearrange input boxes so they are grouped in a less confusing manner.

- Add to userfile as a DET/OVERLAP setting

- Add a tickbox for fit range in the new GUI

- Make it so it works if you enter the same number in both of the fit range boxes.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
It was requested that the user file parsers be updated to allow the merge range to be set from the user file using the syntax. DET/OVERLAP Q1 Q2. 

**To test:**
- Download this data
[loqdemo.zip](https://github.com/mantidproject/mantid/files/1748425/loqdemo.zip)

- In Mantid open the old sans GUI ensure that the instrument is set to LOQ and load the maskfile from the folder. 
- Load the batch file from the folder.
- Go to the reduction settings tab and confirm that the Q range for merge boxes in the bottom right have the values 0.13 and 0.15 and the tickbox is checked.
- Click 1D reduce plot first_time_hab, first_time_main and first_time_merged on a single graph. 
- Confirm that the merged data is identical to the main curve until 0.13, identical to the hab curve above 0.15 and an intermediate value inbetween.
- Open up the new GUI and confirm you get the same graphs if the reduction is run there. 

Partially Fixes #21776 . 

**Release Notes** 
Added to release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/isis_reduction_steps.py [hunk 1]
```diff
@@ -3719,6 +3721,25 @@ def _readFrontRescaleShiftSetup(self, details, reducer):
...
+        rAnds.q_merge_range = False
+        if details.startswith('OVERLAP'):
+            if len(values) == 3:
+                rAnds.q_merge_range = True
+                rAnds.q_min = float(values[1])
+                rAnds.q_max = float(values[2])
+            else:
+                _issueWarning(
+                    "Command: \"DET/" + details + "\" not valid. Expected format is /DET/OVERLAP q1 q2")
+
```
Comment (mantid-roman) [inline scripts/SANS/isis_reduction_steps.py]: If `details` doesn't start with `OVERLAP` the command is silently ignored. Is it intentional?

**Review by mantid-roman — CHANGES_REQUESTED** [review]

**Comment by mantid-roman** [comment]
`Q range for merge` boxes contain 4.0 and 5.0
A number of algorithms report error `XMin must be less than XMax`

**Comment by Mantid-Matthew** [comment]
@mantid-roman I think I have now addressed both these issue, for the first I have removed the if statement which was unneeded as the same check was performed a level above. 

For the second I have fixed some broken bound checking which I think was the cause of the issue.

**Review by mantid-roman — APPROVED** [review]

**Review by mantid-roman — CHANGES_REQUESTED** [review]
Something needs to be checked

**Comment by Mantid-Matthew** [comment]
@mantid-roman I went back and had a double checked on linux and it seemed that it was working with line specific mask files there(so i was wrong on that count) as long as the mask file specified was the same as the original mask file. The difference between the two operating systems was I think down to a  path comparison which was evaluating to false on windows when it should have evaluated to true. I've updated now to hopefully fix that.

**Review by mantid-roman — APPROVED** [review]

**Review by martyngigg — COMMENTED** [review]
Should there be some additional tests here?

**Comment by martyngigg** [comment]
Note: At this point I'd be happy with a new issue/PR for those so that this an make the release.

**Comment by Mantid-Matthew** [comment]
@martyngigg Yes should have some unit tests associated with the change. I have created an issue to add them #21911

**Pull request merged**